src/main/kotlin/com/github/pool_party/pull_party_bot/commands/PingCommandHandlers.kt

```diff
@@ -173,15 +173,25 @@ private fun Bot.handlePartyChangeRequest(isNew: Boolean, msg: Message, args: Str
         return
     }
 
-    val partyName = parsedList[0].removePrefix("@")
+    val partyName = parsedList[0].replace("@", "")
+
+    if (partyName.isBlank()) {
+        sendMessage(chatId, ON_PARTY_NAME_FAIL)
+        return
+    }
 
-    // TODO name validation
     if (!modifyCommandAssertion(chatId, partyName)) {
         return
     }
 
     val users = parsedList.drop(1)
```

Comment:
@nothingelsematters add some sequence magic there, if it would change something

Reply:
It is better to use sequences, because there are lazy. You can think about it as one traversal, but here you create 3 different lists, what for? You can just wrap it in `asSequence()` and `toList()` and it will fly 🚀 .